x-pack/legacy/plugins/code/server/distributed/resource_locator.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Request } from 'hapi';
+import { RequestContext } from './service_definition';
+
+export interface Endpoint {
```

Comment:
Looks like `Endpoint` and `RequestContext` are pretty much the same thing based on this PR. The implementations of `toContext()` of each endpoints can be moved into their constructors.

Reply:
You are right, the endpoint was designed to represent the target kibana node address we are sending to. Here in Local and Multi-node mode, the address is always the same. So, they look like the same thing.